include/zephyr/storage/flash_map.h

```diff
@@ -63,6 +63,8 @@ struct flash_area {
 	size_t fa_size;
 	/** Backing flash device */
 	const struct device *fa_dev;
+	/** Partition label if defined as a property. Otherwise the default is `unlabeled-<fa_id>` */
+	const char *fa_label;
```

Comment:
I split it off into a separate global structure, which allows the compiler to optimize it away if it's not used. I added a separate function `flash_area_label` to flash_map.h, which wraps access to this structure. It's currently only used from the `flash_map list` command.
Since this command is only built if `CONFIG_FLASH_MAP_SHELL` is `y`, this function and the structure containing the runtime instances of the labels are optimized away if it is `n`. 

I've tested it with `rom_report` and with my partition table containing 6 partitions with labels of varying length it does save about 200 rom bytes.